tasks/main.yml

```diff
@@ -1,4 +1,20 @@
 ---
+
+- name: Ensure nix user exists
+  user:
+    name: "{{ nix_user }}"
+    state: present
+  register: nix_user_details
+
+- name: Ensure nix user config dir exists
+  become: true
+  become_user: "{{ nix_user }}"
+  file:
+    # TODO: proper XDG compliance...
+    path: "{{ nix_user_details.home + '/.config' }}/nix"
+    state: directory
```

Comment:
Also, can you clarify what is needed to solve the `TODO`? What is needed for XDG compliance?

Reply:
Hard to give a full answer (I'm not an expert), but at least I'd say if `XDG_CONFIG_DIR` is set it should be used instead of that dir
(see [XDG specs](https://specifications.freedesktop.org/basedir-spec/latest/ar01s03.html))

I tried but env vars with `become` is proving so difficult (for me?) that I gave up